reviews/2020-flux-annual.md

```diff
@@ -0,0 +1,145 @@
+Delivery](https://www.cncf.io/blog/2020/06/12/introducing-the-cncf-technology-radar/),
+meaning it was widely adopted and recommended among the CNCF end user
+community.
+
+There are some proxy metrics for adoption:
+
+ * the #flux channel in CNCF Slack now has >1100 users;
+ * GitHub stars have gone from about 2500 to >5300 in the last year;
+ * we believe (based on optional telemetry in fluxd) that the install
+   base is on the order of tens of thousands of instances.
+
+### Performance against goals
+
+This is the Flux project's first review, so there are no goals already
+on record. However, there are some measures of progress:
+
+ * We have been talking to users and customers a lot. The Flux
+   community was instrumental in running the [GitOps
+   Days](https://www.gitopsdays.com/) event, for example.
+ * We started experimentation in the GitOps Toolkit, designed the APIs
+   and wrote the roadmap together with our community. This move [has
+   been announced](https://www.weave.works/blog/gitops-with-flux-v2)
+   (and [on the mailing
+   list](https://lists.cncf.io/g/cncf-flux-dev/message/26)). The
+   components aim to be generally re-usable and
+   e.g. [Flagger](https://flagger.app/) plans to use the Toolkit’s
+   notification-controller.
+
+One thing that did not go well: Weaveworks announced the collaboration
+of Flux with the Argo project last year. The engineering teams were in
+serious discussions - but at some point it became clear that for the
+Argo project, their GitOps Engine was a natural step in their
+evolution, but for Flux it would simply make it redundantly more like
+ArgoCD (and break backward-compatibility for Flux users). However, the
+silver lining was that in seeking our own way forward, the GitOps
+Toolkit was born.
+
+## Current goals
+
+The main focus for the team is to build out the GitOps Toolkit, which
+will form the basis for Flux v2 and Helm Operator v2.
+
+Once we reach feature parity, we will ask our community of users to
+cut over to using v2 (based on the GOTK) within 6 months.
+
+### How can the CNCF help us achieve our upcoming goals?
+
+We are struggling somewhat with supporting users on Slack and Github
+issues.
+
+ * We recently closed the #flux channel on the Weave Slack as we have
+   had the channel in CNCF Slack open for about a year anyway.
+ * We are also experimenting with GitHub Discussions for
+   [fluxcd/toolkit](https://github.com/fluxcd/toolkit/discussions) --
+   this might help redirect some enquiries to a better venue.
+ * Still it’d be good to learn more about how other CNCF projects are
+   dealing with this by-product of their success.
+
+## Does Flux meet the criteria for incubation?
+
+We think so.
+
+ * Adoption continues to grow, and we see more and more talks, how-to
+   guides and integrations around.
+ * Cloud vendors and industry partners integrate with Flux and are
+   watching GOTK development closely.
+ * As Flux is in a very interesting phase of its development right
```

Comment:
Could you elaborate on "some while" - is it months, years? Or is it something that can't be estimated at this moment.

Reply:
Low numbers of months. There's a timeline [here](https://toolkit.fluxcd.io/roadmap/); we're operating a migration window during which both Flux v1 and v2 are maintained, but expect contributions to Flux v1 to dwindle quickly.